src/Cli/dotnet/ReleasePropertyProjectLocator.cs

```diff
@@ -0,0 +1,153 @@
+                    }
+                    catch (GracefulException)
+                    {
+                        string potentialSln = Directory.GetFiles(arg, "*.sln", SearchOption.TopDirectoryOnly).FirstOrDefault();
+
+                        if (!string.IsNullOrEmpty(potentialSln))
+                            return GetSlnProject(potentialSln, slnProjectPropertytoCheck);
+                    } // If nothing can be found: that's caught by MSBuild XMake::ProcessProjectSwitch -- don't change the behavior by failing here. 
+                }
+            }
+
+            return string.IsNullOrEmpty(potentialProject) ? null : TryGetProjectInstance(potentialProject);
+        }
+
+        /// <returns>The top-level project (first if multiple exist) in a SLN. Returns null if no top level project. Throws exception if two top level projects disagree
+        /// in the configuration property to check.</returns>
+        public override ProjectInstance GetSlnProject(string slnPath, string slnProjectConfigPropertytoCheck = "")
+        {
+            if (Environment.GetEnvironmentVariable("ENABLE_P_RELEASE_SLN") == null) // This has a performance overhead so don't do this unless opted in.
+                return null; // The user will be warned if they do not have this set and try this scenario with one of the properties set.
+
+            SlnFile sln;
+            try
+            {
+                sln = SlnFileFactory.CreateFromFileOrDirectory(slnPath);
+            }
+            catch (GracefulException)
+            {
+                return null; // This can be called if a solution doesn't exist. MSBuild will catch that for us.
+            }
+
+            List<ProjectInstance> configuredProjects = new List<ProjectInstance>();
+            HashSet<string> configValues = new HashSet<string>();
+            bool shouldReturnNull = false;
+
+            Parallel.ForEach(sln.Projects.AsEnumerable(), (project, state) =>
+            {
+                const string topLevelProjectOutputType = "Exe"; // Note that even on Unix when we don't produce exe this is still an exe, same for ASP
+                const string solutionFolderGuid = "{2150E333-8FDC-42A3-9474-1A3956D46DE8}";
+                const string sharedProjectGuid = "{D954291E-2A0B-460D-934E-DC6B0785DB48}";
+
+                if (project.TypeGuid == solutionFolderGuid || project.TypeGuid == sharedProjectGuid || !IsValidProjectFilePath(project.FilePath))
+                    return;
+
+                var projectData = TryGetProjectInstance(project.FilePath);
+                if (projectData == null)
+                    return;
+
+                if (projectData.GetPropertyValue("OutputType") == topLevelProjectOutputType)
+                {
+                    if (ProjectHasUserCustomizedConfiguration(projectData))
+                    {
+                        shouldReturnNull = true;
```

Comment:
Not currently. I don't think that is the desired behavior. If a user provides `-p:Configuration=Blah` but also opted in to `PublishRelease` by default, I don't think it would be a great experience for them to recieve a warning for providing `Blah` after opting into that default.